### PR TITLE
feat: Automated Spell-check for docs and associated spelling fixes

### PR DESCRIPTION
**Pull request by SeanKilleen**
GitHub Issue (If applicable): Supports #7422.

## PR Type

What kind of change does this PR introduce?

- Build or CI related changes
- Documentation content changes

## What is the current behavior?

* Spellchecks are done manually, by humans. Errors can slip by in PRs.
* Some spelling errors possibly exist in the docs

## What is the new behavior?

* Spellchecking is done via GitHub Actions and cSpell. 
* Anyone using VSCode (or GitHub Codespaces, etc.) that has the spellcheck extension will get automatic highlighting and fix suggestions.
* All possible spelling errors reviewed and dictionary updated or fixes made.
* Dev docs updated to reflect how to use the tooling locally.

ℹ️ You can choose to require this as a check to pass (which I suggest doing upon merge of this PR)

## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] Docs have been added/updated which fit [documentation template](https://github.com/unoplatform/uno/blob/master/doc/.feature-template.md) (for bug fixes / features)
- [ ] [Unit Tests and/or UI Tests](https://github.com/unoplatform/uno/blob/master/doc/articles/uno-development/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [ ] Validated PR `Screenshots Compare Test Run` results.
- [x] Contains **NO** breaking changes
- [x] Associated with an issue (GitHub or internal) and uses the [automatic close keywords](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue).
- [ ] Commits must be following the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) specification.

## Other information

Internal Issue (If applicable):

## Remaining / Surfaced tasks 

(so they don't get lost)

* [x] Fix API "favourites" routes
* [x] Update regex to more inclusive regex
* [x] Split added vs ignored words in cSpell config
* [x] YAML formatting consistent with rest of code

### CODE REVIEW

**Comment by SeanKilleen** [comment]
ℹ️ Note that someone will need to enable the workflows I'm adding here in order to verify that they're working correctly. If you review the YAML file, you'll see it's pretty straightforward.

**Comment by SeanKilleen** [comment]
At start, 699 issues found (though much of that will be about adding known words to the dictionary).

**Comment by SeanKilleen** [comment]
@jeromelaban -- saw that you marked this PR as draft. Could you please click to approve the workflow to run so that I can ensure things are working as expected with the GitHub Action? (at your leisure of course.)

File: doc/articles/guides/silverlight-migration/12-migrate-auth-ui.md
Comment (Youssef1313) [inline doc/articles/guides/silverlight-migration/12-migrate-auth-ui.md:91]:
```suggestion
1. To add the login form string resources, open the `Strings\en\Resources.resw` file and add the following strings:
```

File: doc/articles/guides/silverlight-migration/12-migrate-auth-ui.md
Comment (Youssef1313) [inline doc/articles/guides/silverlight-migration/12-migrate-auth-ui.md:131]:
```suggestion
1. To add the input controls string resources, open the `Strings\en\Resources.resw` file and add the following strings:
```

File: doc/articles/guides/silverlight-migration/05-string-resources.md
Comment (SeanKilleen) [reply doc/articles/guides/silverlight-migration/05-string-resources.md:85]: Thanks for these! Wasn't sure if that was important to the original so didn't want to change it without confirmation which you've given me proactively. :)

File: doc/articles/howto-consume-webservices.md
Comment (jeromelaban) [inline doc/articles/howto-consume-webservices.md:1417]: This should not be changed, as well as the other references below

**Review by jeromelaban — COMMENTED** [review]
This is looking very nice! Thanks for your contribution. We'll enable the additional action as a requirement, once merged.

File: cSpell.json
Comment (Youssef1313) [inline cSpell.json:39]: Can you use the regex specified [here](https://cspell.org/configuration/patterns/) instead? It's a more accurate regex.

File: doc/articles/howto-consume-webservices.md
Comment (SeanKilleen) [reply doc/articles/howto-consume-webservices.md:1417]: Right, makes sense 👍 thanks! I'll fix this prior to review.

**Comment by SeanKilleen** [comment]
Down to 94 reported issues.

File: .github/workflows/docs.yml
Comment (SeanKilleen) [reply .github/workflows/docs.yml:25]: Ah yes, I'll undo that, thanks -- it was an auto-formatting issue.

Comment (Youssef1313) [reply .github/workflows/docs.yml:25]: Probably because we don't have `indent_size` in `.editorconfig` for yml files.

FYI @jeromelaban

**Comment by SeanKilleen** [comment]
OK -- at this point, my local cspell reports no errors. 🎉 

I'm going to move on to making the suggested changes and resolving any build errors.

We still do not know if the GitHub action is working as defined because it has not yet been enabled as far as I can tell.

File: cSpell.json
Comment (SeanKilleen) [reply cSpell.json]: Done! Also due to your tip I learned about cSpell's ability to name patterns and then ignore based on name, which I like and may use elsewhere.

File: .github/workflows/docs.yml
Comment (SeanKilleen) [reply .github/workflows/docs.yml]: ✅ I added the editorconfig value and re-formatted the YAML file.

File: cSpell.json
Comment (Youssef1313) [inline cSpell.json:110]: Consider either updating or removing the link

File: doc/articles/features/routed-events.md
Comment (Youssef1313) [inline doc/articles/features/routed-events.md:74]: unmanaged is commonly used without hyphen.

File: doc/articles/supported-features.md
Comment (Youssef1313) [inline doc/articles/supported-features.md:49]: This change might be incorrect. @jeromelaban could you confirm?

File: doc/articles/features/uno-cupertino.md
Comment (SeanKilleen) [reply doc/articles/features/uno-cupertino.md]: I'm happy to defer to others here. For now I've reverted these changes and added to the dictionary. 👍

**Comment by SeanKilleen** [comment]
@Youssef1313 @jeromelaban I noticed that my commits are definitely going to fail the conventional commits validation.

After you're reviewed the individual changes, I'm happy to squash and force push my changes with a conventional commit title.

I assumed this would be squash-merged at the end but didn't realize the commit titles would prevent CI from running.

File: .editorconfig
Comment (SeanKilleen) [inline .editorconfig:33]: ℹ️ I added this per discussion. If you'd like it to be a separate commit, I can remove and do a separate PR; totally fine either way.

File: cSpell.json
Comment (SeanKilleen) [inline cSpell.json:78]: ℹ️ Quick note: `words` can become spellCheck suggestions, whereas `ignoreWords` will not.

File: doc/articles/Uno-UI-Performance.md
Comment (SeanKilleen) [inline doc/articles/Uno-UI-Performance.md:3]: ℹ️ Apologies. Looks like my auto-formatting got this file. If you want me to revert the formatting changes throughout, just let me know.

File: doc/articles/guides/hotswap-app-language.md
Comment (SeanKilleen) [inline doc/articles/guides/hotswap-app-language.md:144]: ℹ️ FYI: my approach to the french translations here was to use back-ticks on them, as it seemed reasonable enough while not requiring us to add them to dictionaries or update the dictionary to include the french language.

File: doc/articles/get-started-next-steps.md
Comment (SeanKilleen) [inline doc/articles/get-started-next-steps.md:22]: ℹ️ This is miscellaneous but I wanted to include it for Skye's sake, since I know her. Skye transitioned from Andrew and her proper name is reflected elsewhere in these docs; looks like it just got missed here.

File: doc/articles/get-started-next-steps.md
Comment (SkyeHoefling) [inline doc/articles/get-started-next-steps.md:42]:
```suggestion
_Skye is also currently writing a book on Uno Platform; we will add it here upon publication._
```

**Review by SkyeHoefling — CHANGES_REQUESTED** [review]
Thank you so much @SeanKilleen for updating my name in this PR. I noticed there was 1 place that was missed and I left a code suggestion

**Review by Youssef1313 — APPROVED** [review]
Not a detailed review, but LGTM overall.

**Comment by SeanKilleen** [comment]
Thanks for the approval @Youssef1313! 

Given that someone has approved it, I am going to now squash my changes and force-push so that I can use a conventional commit and ensure the rest of the pipelines are passing too.

File: doc/articles/supported-features.md
Comment (MartinZikmund) [reply doc/articles/supported-features.md:49]: I would keep inlines here, as it matches the XAML `Inlines` property

**Review by MartinZikmund — APPROVED** [review]

File: cSpell.json
Comment (Youssef1313) [inline cSpell.json:124]: The failure in CI is because this regex isn't accurate.
This suggestion makes the regex more accurate.

*Note: If you'll commit from GitHub, don't forget to follow conventional commits.*

```suggestion
			"pattern": "\\`([^\\`\\r\\n]+?)\\`",
```

File: cSpell.json
Comment (Youssef1313) [inline cSpell.json:158]: Why this one is ignored?

Comment (SeanKilleen) [reply cSpell.json:158]: Sorry, I had commented the reason for this but it may have disappeared when I force merged.

I ignored this because of the amount of HTML in file caused many false positives and it seemed better to ignore it as a non-markdown file.

File: doc/articles/supported-features.md
Comment (SeanKilleen) [reply doc/articles/supported-features.md:49]: Sure thing -- I'll revert this change and add it to the dictionary instead 👍

Comment (SeanKilleen) [reply doc/articles/supported-features.md]: ✅ Done.

**Comment by SeanKilleen** [comment]
Hey all -- as far as I can tell:

✅ All comments have been addressed
✅ Commits are looking good and passing
✅ CI is passing or seems about to
✅ I'm confident the GitHub Action will work once enabled in this PR
✅ It runs and reports zero errors in my environment 

Given that, I'd love to do a last pass and get this merged. It's not that I'm impatient; it's that I know this touches a lot of files and I don't want it to go too long before merge conflicts are a thing.

File: doc/articles/guides/hotswap-app-language.md
Comment (SeanKilleen) [reply doc/articles/guides/hotswap-app-language.md:144]: Resolving this, as it has to be resolved for merging.

File: doc/articles/Uno-UI-Performance.md
Comment (SeanKilleen) [reply doc/articles/Uno-UI-Performance.md:3]: Resolving this; as it had to be resolved for merging.

**Comment by SeanKilleen** [comment]
@MartinZikmund / @jeromelaban / @Youssef1313 -- just checking on what the path to merge here is. It looks like all the comments have been addressed and I'm happy to tackle anything else. 

I won't ping more on this because I don't want to disturb you further 👍 but wanted to get this in before other PRs cause merge conflicts. given the number of files affected..

File: doc/articles/wpf-migration.md
Comment (jeromelaban) [inline doc/articles/wpf-migration.md:11]: `bytecode` is a commonly used spelling, we can also add it to the exclusions.

**Review by jeromelaban — APPROVED** [review]
This is a very useful PR, thank you for your contribution! Just a few minor fixes and it will be good to merge!

File: doc/articles/wpf-migration.md
Comment (SeanKilleen) [reply doc/articles/wpf-migration.md]: ✅ Done

File: doc/articles/features/working-with-accessibility.md
Comment (SeanKilleen) [reply doc/articles/features/working-with-accessibility.md]: ✅ Done

**Comment by SeanKilleen** [comment]
@jeromelaban changes made 👍 ready to go at your leisure.

**Comment by SeanKilleen** [comment]
@jeromelaban I merged master and resolved a minor conflict. This PR is ready to go at your earliest convenience.

File: doc/articles/contributing/guidelines/breaking-changes.md
Comment (Xiaoy312) [inline doc/articles/contributing/guidelines/breaking-changes.md:53]: from this line till the end, it is an example of output (yes, it is in markdown)
which should be wrapped between:
~~~
```md
```
~~~
and, anything in between should be kept intact

Comment (SeanKilleen) [reply doc/articles/contributing/guidelines/breaking-changes.md:53]: In this case I believe I only modified the spelling of the word, so as to keep to the purpose of the PR. If you'd like me to make an additional change, I'm happy to consider it.

Comment (Xiaoy312) [reply doc/articles/contributing/guidelines/breaking-changes.md:53]: on lines 301 and 370, you added `inline-code-snippet`; these are part of said output
if you could, can you please wrap L53~EOF in code snippet and undo the changes in between those lines? thanks

Comment (SeanKilleen) [reply doc/articles/contributing/guidelines/breaking-changes.md:54]: ✅  @Xiaoy312 thanks for clarifying! I made the requested changes.

**Comment by SeanKilleen** [comment]
Looks like these completed running and the only failures seem unrelated to the changes. 👍

**Comment by jeromelaban** [comment]
> Looks like these completed running and the only failures seem unrelated to the changes. 👍

Indeed, those are unrelated failures, linked to the project getting bigger :) We're fixing those.

**Pull request merged**
